lib/src/execute.rs

```diff
@@ -228,7 +229,7 @@ impl ExecuteCtx {
         remote: IpAddr,
     ) -> Result<(), ExecutionError> {
         info!("handling request {} {}", req.method(), req.uri());
-
+        let now = Instant::now();
```

Comment:
how about renaming this variable ever so slightly, to something like...

```suggestion
        let start_timestamp = Instant::now();
```